docs/learn/smart-contract-developers/deploy-up-with-lsp23.md

```diff
@@ -0,0 +1,447 @@
+    initializationCalldata: '0xc4d66de8', // initialize() function selector ( the deployed Universal Profile address will be appended to this calldata)
+    extraInitializationParams: '0x',
+  };
+}
+```
+
+### Create the Universal Profile initialisation calldata
+
+When deploying your Universal Profile, you might want to initialise it with some data. For example, you may want to set the Universal Profile [LSP3 Metadata](../../standards/universal-profile/lsp3-profile-metadata.md), set the Universal Receiver to the Universal Profile, give some [LSP6 permissions](../../standards/universal-profile/lsp6-key-manager#permissions.md) to some controllers, etc. For the following guide we will:
+
+- set the Universal Profile LSP3 Metadata
+- set the [Universal Receiver](../../standards/generic-standards/lsp1-universal-receiver.md) to the Universal Profile
+- give some permissions to the Universal Receiver ([`SUPER_SETDATA` and `REENTRANCY`](../../standards/universal-profile/lsp6-key-manager#permissions))
+- give all permissions to the main controller
+  set the [`AddressPermissions[]`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-6-KeyManager.md#addresspermissions) length data key
+- set the [`AddressPermissions[]`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-6-KeyManager.md#addresspermissions) array data key
+- set the [`AddressPermissions:Permissions:<address>`](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-6-KeyManager.md#addresspermissionspermissionsaddress) data keys
+
+#### Create the LSP3 Metadata
+
+The LSP3 Metadata is a JSON object that contains information about the Universal Profile. For more information on the LSP3 Metadata, please refer to the [LSP3 Profile Metadata specification](../../standards/universal-profile/lsp3-profile-metadata.md). For the sake of simplicity, we will not create the JSON object in this guide but use random bytes instead:
+
+```typescript
+async function main() {
+  // previous code
+
+  const lsp3DataKey = ERC725YDataKeys.LSP3.LSP3Profile;
+  const lsp3DataValue = ethers.randomBytes(32); // this is just a random value for the LSP3Profile
+}
+```
+
+#### Create the Universal Receiver data key
+
+```typescript
+async function main() {
+  // previous code
+
+  const universalReceiverDataKey =
+    ERC725YDataKeys.LSP1.LSP1UniversalReceiverDelegate;
+}
+```
+
+#### Create the permissions data keys and data values
+
+Let's start by creating the permissions data keys and data values of the Universal Receiver. The Universal Receiver will have the [`SUPER_SETDATA` and `REENTRANCY`](../../standards/universal-profile/lsp6-key-manager#permissions) permissions. At the moment we are writing this guide, the combination of these two permissions is `0x0000000000000000000000000000000000000000000000000000000000060080` however this may change in the future. You can check if this is still the case by looking at the [ERC725 Inspection Tool](https://erc725-inspect.lukso.tech/key-manager).
```

Comment:
Wouldn't it be safer to use _erc725.js_ here instead of a literal constant?

Reply:
sounds good. wiill create a ticket to update this guide using erc725.js because article needs to be released asap